cpp/benchmarks/common/generate_nullmask.cu

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "generate_input.hpp"
+
+#include <cudf/detail/valid_if.cuh>
+#include <cudf/null_mask.hpp>
+
+#include <thrust/random.h>
+
+/**
+ * @brief valid bit generator with given probability [0.0 - 1.0]
```

Comment:
I'd change the docs / naming here. There's no such thing as a "valid" bit -- only set (1) or unset (0) bits. This is layered with our interpretation of a set of bits as a null mask, but this function as written only deals with generating bools.

This mixing of semantics was a large part of what I untangled in https://github.com/rapidsai/cudf/pull/9588. In particular, there were issues with how that interpretation of valid/set and null/unset results in higher-order semantics when a null mask is `nullptr`. It's not possible to count "set" bits in the data pointed to by `nullptr`, but we have a specific interpretation of that in the context of validity / cudf null masks.

I want to emphasize this distinction and keep our lowest-level primitives free of the valid/null interpretations of bits/bools. Including those semantics in functions like `create_random_null_mask` is fine, because the interpretation is explicit.

Reply:
updated.